ui/client/pages/threaddump/threaddump.jsx

```diff
@@ -0,0 +1,105 @@
+
+// Executors Page
+Router.route("/a/:_appId/executors/:_execId/threadDump", {
+  waitOn: function() {
+    return [
+      Meteor.subscribe('app', this.params._appId),
+      Meteor.subscribe('executor-thread-dumps', this.params._appId, this.params._execId)
+    ];
+  },
+  action: function() {
+    this.render("threadDumpPage", {
+      data: {
+        appId: this.params._appId,
+        app: Applications.findOne(),
+        execId: this.params._execId,
+        threads: ExecutorThreadDumps.find().fetch(),
+        executorsTab: 1
+      }
+    });
+  },
+  name: 'threads'
+});
+
+// Thread display box, takes 'thread' and 'isGloballyCollapsed' as props
+ThreadTraceBox = React.createClass({
+  getInitialState() {
+    return {
+      isCollapsed: this.props.isGloballyCollapsed
+    };
+  },
+  componentWillReceiveProps(nextProps) {
+    this.setState({isCollapsed: nextProps.isGloballyCollapsed});
+  },
+  toggleBody() {
+    this.setState({isCollapsed: !this.state.isCollapsed});
+  },
+  render() {
+    var header = "Thread " + this.props.thread.id + ": " + this.props.thread.threadName +
+      " (" + this.props.thread.threadState + ")";
+    return <div className="accordion-group">
+      <div className="accordion-header">
+        <a href="#" onClick={this.toggleBody} className="accordion-toggle">{header}</a>
+      </div>
+      <div className={"accordion-body" + ((this.state.isCollapsed) ? " hidden" : "")}>
+        <pre>{this.props.thread.stackTrace}</pre>
+      </div>
+    </div>;
+  }
+})
+
+// Showing global list of threads for executor. Also displays link to toggle all thread dump stack
+// traces using `isGloballyCollapsed`, this will either collapse or expand stack traces regardless
+// of their state, which will be reset to global.
+ThreadDumpList = React.createClass({
+  getInitialState() {
+    return {
+      isGloballyCollapsed: true
+    };
+  },
+  toggleGlobal() {
+    this.setState({isGloballyCollapsed: !this.state.isGloballyCollapsed})
+  },
+  getLastUpdateTime() {
+    var time = "Undefined";
```

Comment:
Last time is chosen as max update time across all threads for executor. I will update comment on this.